src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -396,8 +396,7 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
             .tcx
             .doc_link_resolutions(module_id)
             .get(&(Symbol::intern(path_str), ns))
-            .copied()
-            .unwrap_or_else(|| panic!("no resolution for {:?} {:?} {:?}", path_str, ns, module_id))
+            .copied()?
```

Comment:
This is not a fix. You've just hidden the bug; this would have caused most of the other bugs that were reported to result in silently broken docs instead of ICEs. Please fix this in rustc_resolve instead.

Reply:
cc https://github.com/rust-lang/rust/pull/95645#discussion_r841839491

Reply:
You're right. I'll put this back as a draft and fix the underlying issue.